scipy/optimize/_linprog_util.py

```diff
@@ -231,62 +262,50 @@ def _clean_inputs(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None, bounds=None,
             raise ValueError(
                 "Invalid input for linprog: A_eq must not contain values "
                 "inf, nan, or None")
-    except TypeError:
-        raise TypeError(
-            "Invalid input for linprog: A_eq must be a 2D array with each "
-            "row representing an equality constraint")
 
     try:
-        try:
-            b_eq = np.array(
-                [], dtype=float) if b_eq is None else np.asarray(
-                b_eq, dtype=float).copy().squeeze()
-        except BaseException:
-            raise TypeError
-        if b_eq.size == 1:
-            b_eq = b_eq.reshape((-1))
-        if len(b_eq.shape) != 1:
+        b_eq = _format_b_constraints(b_eq)
+    except ValueError:
+        raise TypeError(
+            "Invalid input for linprog: b_eq must be a 1D array of "
+            "numerical values, each representing the upper bound of an "
+            "inequality constraint (row) in A_ub")
+    else:
+        if b_eq.shape != (n_eq,):
             raise ValueError(
                 "Invalid input for linprog: b_eq should be a 1D array; it "
-                "must not have more than one non-singleton dimension")
-        if len(b_eq) != n_eq:
-            raise ValueError(
-                "Invalid input for linprog: the number of rows in A_eq "
-                "must be equal to the number of values in b_eq")
+                "must not have more than one non-singleton dimension and "
+                "the number of rows in A_eq equal to the number of values "
+                "in b_eq")
         if not(np.isfinite(b_eq).all()):
             raise ValueError(
                 "Invalid input for linprog: b_eq must not contain values "
                 "inf, nan, or None")
-    except TypeError:
-        raise TypeError(
-            "Invalid input for linprog: b_eq must be a 1D array of "
-            "numerical values, each representing the right hand side of an "
-            "equality constraints (row) in A_eq")
 
+    # x0 gives a (optional) starting solution to the solver. As this is
+    # not required if x0 is None no initial solution is used and does not
```

Comment:
"As this is not required if x0 is None no initial solution is used and does not require checks."
Consider:
"If x0 is None, skip the checks. Initial solution will be generated automatically."